hyperspy/axes.py

```diff
@@ -269,20 +269,11 @@ class BaseDataAxis(t.HasTraits):
     """
     name = t.Str()
     units = t.Str()
-    size = t.CInt()
-    low_value = t.Float()
-    high_value = t.Float()
-    value = t.Range('low_value', 'high_value')
-    low_index = t.Int(0)
-    high_index = t.Int()
-    slice = t.Instance(slice)
+
     navigate = t.Bool(False)
     is_binned = t.Bool(False)
-    index = t.Range('low_index', 'high_index')
```

Comment:
`index` or `_index` still need to be a trait to be synchronise with the GUI widgets.

Reply:
I have the index as a trait for the DataAxis and not for the BaseDataAxis.  Is that okay?  My idea is that the BaseDataAxis would only ever be used if you are trying to define something like a ragged dimension.  In which case the index doesn't have a physical sense.  I can set it as <undefined> if that works better

Reply:
Sorry, it would have been less confusing if my comment was in a better place. What I meant is that the `index` or `_index` needs to be a `traiits` type, e.g. an instance of `traits.api.Range`.

Reply:
Ahh, that makes more sense.  That requires that the low and high indexes be traits as well. Or at least to my understanding that is the case.  I currently have it so that the index is set with a `@index.setter` function which won't update outside of the index high and low.  I'm not terribly familiar with how else this might interact with the API.  I can make the change fairly easily.

Reply:
> Ahh, that makes more sense. That requires that the low and high indexes be traits as well. Or at least to my understanding that is the case.

Yes, this is correct.

> I currently have it so that the index is set with a `@index.setter` function which won't update outside of the index high and low. I'm not terribly familiar with how else this might interact with the API.

This is a details, which will be to sorted in this PR and we can come back to it toward the end of this PR when other things are settled.

Reply:
So this runs into some complications as t.Range needs to be set with a value and not a Property initially.

We could do this:

```python
class DataAxis:
    axis = t.Array()  # Axis added as trait to track changes to axis
    high_index = t.Int(1)
    low_index = 0
    _index = t.Range(low_index, high_index)

    @observe("axis")
    def update_high_index(self, event):
        self.high_index = self.size-1
```

Or keep it the way it is.  I don't think it is too confusing right now but either works!